## Part 1: A Quick Introduction ##
1. 20 is printed by line 9
2. 20 is printed by line 13
3. 20 is printed by line 9
4. error is given by line 13 because result is a type let variable, and it blocks access outside of the block. We can see that line 13 is declared outside of the block where result is declared 
5. error should be produced because we are reassigning the variable result, which is not allowed for a constant variable.
6. the same answer as question number 5
   